packages/main/src/files/request.ts

```diff
@@ -131,13 +131,23 @@ export async function makeFilesRequest(
 }
 
 /**
- * Get AbortSignal if timeout is specified
+ * Create an AbortSignal based on the timeout and abortSignal in the
+ * RequestOptions.
  */
 function getSignal(requestOptions?: RequestOptions): AbortSignal | null {
-  if (requestOptions?.timeout >= 0) {
-    const abortController = new AbortController();
-    const signal = abortController.signal;
-    setTimeout(() => abortController.abort(), requestOptions.timeout);
-    return signal;
+  if (
+    requestOptions?.abortSignal !== undefined ||
+    requestOptions?.timeout >= 0
+  ) {
+    const controller = new AbortController();
+    if (requestOptions?.timeout >= 0) {
+      setTimeout(() => controller.abort(), requestOptions.timeout);
+    }
+    if (requestOptions.abortSignal) {
```

Comment:
For reference, here's the previous PR and the discussion on it: https://github.com/google-gemini/generative-ai-js/pull/50

I think we can just keep the previous "signal" in our code (and name it `timeoutSignal`) and then we can just `return AbortSignal.any([timeoutSignal, userSignal])`. We may also want to name the user-provided signal field just "signal" so that it matches the native API, indicating to the user that we're just passing it through with no modifications.

Reply:
`AbortSignal.any()` requires node v20.3.0, and has only made it into Chrome last fall, and Safari this March. 

Currently our package.json engines directive is `">=18.0.0"`.

Reply:
> We may also want to name the user-provided signal field just "signal" so that it matches the native API, indicating to the user that we're just passing it through with no modifications.

Done!